### PR TITLE
Printable/v13

### PR DESCRIPTION
**Pull request by victorjulien**
https://redmine.openinfosecfoundation.org/issues/6553

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1493

replaces #9854
- rebase
- reimplement frame logging
- membuffer cleanups
- minor fixes and cleanups

### CODE REVIEW

File: src/output-json-frame.c [hunk 1]
```diff
@@ -125,61 +125,90 @@ static void PayloadAsHex(const uint8_t *data, uint32_t data_len, char *str, size
...
-    jb_set_base64(js, "payload", data, data_len);
+    struct FrameJsonStreamDataCallbackData cbd = {
+        .payload = buffer, .frame = frame, .last_re = frame->offset
+    };
+    uint64_t unused = 0;
+    StreamReassembleLog(
+            ssn, stream, FrameJsonStreamDataCallback, &cbd, frame->offset, &unused, false);
+    /* if we have all data, but didn't log until the end of the frame, we have a gap at the
+     * end of the frame
+     * TODO what about not logging due to buffer full? */
```
Comment (victorjulien) [inline src/output-json-frame.c:197]: could perhaps end with something like `[3124 more bytes]`

Comment (victorjulien) [reply src/output-json-frame.c:197]: or simpler, just something like `[...]`

**Comment by victorjulien** [comment]
Some examples:
https://github.com/OISF/suricata-verify/pull/1493/files#diff-aeec7dad229d116ae9cf51d827b8eb996531daf94c75fed3a87aa58a5c033575R30

https://github.com/OISF/suricata-verify/pull/1493/files#diff-6e0eb101fa5e4a71a07ef0391a448fc47528de8efc98d5a93f4115ebbbcde938R27

https://github.com/OISF/suricata-verify/pull/1493/files#diff-3e627fb7bf9154c875467fc406246d519b9fdf5b58e9b4614d0f90be75adf076R50

https://github.com/OISF/suricata-verify/pull/1493/files#diff-c916c1b31d68706709aef64076b156d7850a89a2eb49bfd458ebb9147a34db39R15

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 16719

**Comment by victorjulien** [comment]
Replaced by #10261

**Pull request closed without merging**